Mathlib/MeasureTheory/Measure/CharacteristicFunction.lean

```diff
@@ -68,4 +91,104 @@ theorem ext_of_integral_char_eq (he : Continuous e) (he' : e ≠ 1)
 
 end ext
 
+section InnerProductSpace
+
+variable {E : Type*} {mE : MeasurableSpace E} {μ : Measure E} {t : E}
+
+/-- The characteristic function of a measure in an inner product space. -/
+noncomputable def charFun [Inner ℝ E] (μ : Measure E) (t : E) : ℂ := ∫ x, exp (⟪x, t⟫ * I) ∂μ
+
+lemma charFun_apply [Inner ℝ E] (t : E) : charFun μ t = ∫ x, exp (⟪x, t⟫ * I) ∂μ := rfl
+
+lemma charFun_apply_real {μ : Measure ℝ} (t : ℝ) :
+    charFun μ t = ∫ x, exp (t * x * I) ∂μ := by simp [charFun_apply]
+
+variable [SeminormedAddCommGroup E] [InnerProductSpace ℝ E]
+
+@[simp]
+lemma charFun_zero (μ : Measure E) : charFun μ 0 = (μ Set.univ).toReal := by
+  simp [charFun_apply]
+
+@[simp]
+lemma charFun_zero_measure : charFun (0 : Measure E) t = 0 := by simp [charFun_apply]
+
+@[simp]
+lemma charFun_neg (t : E) : charFun μ (-t) = conj (charFun μ t) := by
+  simp [charFun_apply, ← integral_conj, ← exp_conj]
+
+/-- `charFun` as the integral of a bounded continuous function. -/
+lemma charFun_eq_integral_innerProbChar : charFun μ t = ∫ v, innerProbChar t v ∂μ := by
+  simp [charFun_apply, innerProbChar_apply]
+
+lemma charFun_eq_integral_probChar (y : E) : charFun μ y = ∫ x, (probChar ⟪x, y⟫ : ℂ) ∂μ := by
```

Comment:
nitpick: in the lemmas before and after this one, you use `t` instead of `y` as the argument